components/browser/engine-gecko-beta/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineViewTest.kt

```diff
@@ -104,4 +105,24 @@ class GeckoEngineViewTest {
         verify(geckoView).releaseSession()
         verify(engineSession).unregister(any())
     }
+
+    @Test
+    fun `View will rebind session if session crashed`() {
+        val engineView = GeckoEngineView(context)
+        val engineSession = mock<GeckoEngineSession>()
+        val geckoSession = mock<GeckoSession>()
+        val geckoView = mock<NestedGeckoView>()
+
+        whenever(engineSession.geckoSession).thenReturn(geckoSession)
+        engineView.currentGeckoView = geckoView
+
+        engineView.render(engineSession)
+
+        Mockito.reset(geckoView)
```

Comment:
I'm not completely sure we want to reset the mock here?

Reply:
The initial render call will already call `setSession()` and I want to just ignore that and start fresh to only verify what happens when calling `onProcessKilled()`.

Reply:
👍